pygmt/tests/test_grd2cpt.py

```diff
@@ -0,0 +1,316 @@
+"""
+Tests for grd2cpt.
+"""
+import os
+
+import numpy as np
+import pytest
+from pygmt import Figure, grd2cpt
+from pygmt.datasets import load_earth_relief
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import GMTTempFile
+from pygmt.helpers.testing import check_figures_equal
+
+TEST_DATA_DIR = os.path.join(os.path.dirname(__file__), "data")
+POINTS_DATA = os.path.join(TEST_DATA_DIR, "points.txt")
+
+
+@pytest.fixture(scope="module", name="points")
+def fixture_points():
+    """
+    Load the points data from the test file.
+    """
+    return np.loadtxt(POINTS_DATA)
+
+
+@pytest.fixture(scope="module", name="region")
+def fixture_region():
+    """
+    The data region.
+    """
+    return [10, 70, -5, 10]
+
+
+@pytest.fixture(scope="module", name="grid")
+def fixture_grid():
+    """
+    Load the grid data from the sample earth_relief file.
+    """
+    return load_earth_relief(registration="gridline")
+
+
+@check_figures_equal()
+def test_grd2cpt(grid):
+    """
+    Test creating a CPT with grd2cpt to create a CPT based off a grid input and
+    plot it with a color bar.
+    """
+    fig_ref, fig_test = Figure(), Figure()
+    # Use single-character arguments for the reference image
+    fig_ref.basemap(B="a", J="W0/15c", R="d")
+    grd2cpt(grid=grid)
+    fig_ref.colorbar(B="a2000")
+    fig_test.basemap(frame="a", projection="W0/15c", region="d")
+    grd2cpt(grid=grid)
+    fig_test.colorbar(frame="a2000")
+    return fig_ref, fig_test
+
+
+@check_figures_equal()
+def test_grd2cpt_to_plot_points(points, region, grid):
+    """
+    Use CPT to change color of points.
+    """
+    fig_ref, fig_test = Figure(), Figure()
+    # Use single-character arguments for the reference image
+    fig_ref.basemap(R=region, J="X15c", B="a")
+    grd2cpt(grid=grid, C="rainbow")
+    fig_ref.plot(
+        x=points[:, 0],
+        y=points[:, 1],
+        G=points[:, 2],
+        R=region,
+        S="c1c",
+        C=True,
+    )
```

Comment:
```suggestion
    fig_ref.plot(
        x=points[:, 0], y=points[:, 1], G=points[:, 2], R=region, S="c1c", C=True
    )
```
This can fit on one line.